packages/db/src/models/granule-duplicates.ts

```diff
@@ -0,0 +1,97 @@
+import { Knex } from 'knex';
+
+import { isRecordDefined } from '../database';
+import { TableNames } from '../tables';
+
+import { PostgresGranuleDuplicate } from '../types/granule-duplicate';
+
+export default class GranuleDuplicatesPgModel {
+  readonly tableName: TableNames;
+
+  // can't extend base class because type for this data doesn't contain
+  // an updated_at property
+  constructor() {
+    this.tableName = TableNames.granuleDuplicates;
+  }
+
+  async create(
+    knexTransaction: Knex | Knex.Transaction,
+    item: PostgresGranuleDuplicate
+  ) {
+    return await knexTransaction(this.tableName).insert(item);
+  }
+
+  async exists(
+    knexTransaction: Knex | Knex.Transaction,
+    item: PostgresGranuleDuplicate
+  ) {
+    return isRecordDefined(await knexTransaction(this.tableName).where(item).first());
+  }
+
+  async upsert(
+    knexTransaction: Knex | Knex.Transaction,
+    item: PostgresGranuleDuplicate
+  ) {
+    return await knexTransaction(this.tableName)
+      .insert(item)
+      .onConflict(['granule_cumulus_id', 'group_id'])
+      .merge()
+      .returning('*');
+  }
+  /**
+   * Creates multiple granuleExecutions in Postgres
+   *
+   * @param {Knex | Knex.Transaction} knexOrTransaction - DB client or transaction
+   * @param {PostgresGranuleDuplicate[]} items - Records to insert into the DB
+   * @param {string | Array<string>} returningFields - A string or array of strings
+   *   of columns to return. Defaults to 'cumulus_id'.
+   * @returns {Promise<PostgresGranuleDuplicate[]>} Returns an array of objects
+   *   from the specified column(s) from returningFields.
+   */
+  async insert(
+    knexOrTransaction: Knex | Knex.Transaction,
+    items: PostgresGranuleDuplicate[],
+    returningFields: string | string[] = '*'
+  ): Promise<PostgresGranuleDuplicate[]> {
+    return await knexOrTransaction(this.tableName)
+      .insert(items)
+      .returning(returningFields);
+  }
+
+  async delete(
+    knexTransaction: Knex.Transaction,
+    params: Partial<PostgresGranuleDuplicate>
+  ): Promise<number> {
+    return await knexTransaction(this.tableName)
+      .where(params)
+      .del();
```

Comment:
This is probably what we want but, just to be obnoxious, what happens if a user deletes a record for the "active" granule?

Reply:
The model doesn't necessarily need validation, even if there's a case for it (not sure). The endpoint would probably handle some of the more complex tasks in re-assigning active status, etc.?